File: app/src/androidTest/java/org/tasks/makers/GoogleTaskMaker.java
package org.tasks.makers;

import static com.natpryce.makeiteasy.Property.newProperty;
import static org.tasks.makers.Maker.make;

import com.natpryce.makeiteasy.Instantiator;
import com.natpryce.makeiteasy.Property;
import com.natpryce.makeiteasy.PropertyValue;
import com.todoroo.astrid.helper.UUIDHelper;
import org.tasks.data.GoogleTask;

public class GoogleTaskMaker {

  public static final Property<GoogleTask, String> LIST = newProperty();
  public static final Property<GoogleTask, Integer> ORDER = newProperty();
  public static final Property<GoogleTask, String> REMOTE_ID = newProperty();
  public static final Property<GoogleTask, Integer> TASK = newProperty();

  private static final Instantiator<GoogleTask> instantiator = lookup -> {
    GoogleTask task = new GoogleTask();
    task.setListId(lookup.valueOf(LIST, "1"));
    task.setOrder(lookup.valueOf(ORDER, 0));
    task.setRemoteId(lookup.valueOf(REMOTE_ID, UUIDHelper.newUUID()));
    task.setTask(lookup.valueOf(TASK, 1));
    return task;
  };

  @SafeVarargs
  public static GoogleTask newGoogleTask(PropertyValue<? super GoogleTask, ?>... properties) {
    return make(instantiator, properties);
  }
}

File: app/src/test/java/org/tasks/makers/GoogleTaskMaker.kt
package org.tasks.makers

import com.natpryce.makeiteasy.Instantiator
import com.natpryce.makeiteasy.Property
import com.natpryce.makeiteasy.Property.newProperty
import com.natpryce.makeiteasy.PropertyValue
import com.todoroo.astrid.helper.UUIDHelper
import org.tasks.makers.Maker.make

object GoogleTaskMaker {
    val LIST: Property<GoogleTask, String> = newProperty()
    val ORDER: Property<GoogleTask, Long> = newProperty()
    val REMOTE_ID: Property<GoogleTask, String> = newProperty()
    val TASK: Property<GoogleTask, Long> = newProperty()
    val PARENT: Property<GoogleTask, Long> = newProperty()
    val REMOTE_PARENT: Property<GoogleTask, String?> = newProperty()

    private val instantiator = Instantiator<GoogleTask> {
        val task = GoogleTask()
        task.calendar = it.valueOf(LIST, "1")
        task.order = it.valueOf(ORDER, 0)
        task.remoteId = it.valueOf(REMOTE_ID, UUIDHelper.newUUID())
        task.task = it.valueOf(TASK, 1)
        task.parent = it.valueOf(PARENT, 0L)
        task.remoteParent = it.valueOf(REMOTE_PARENT, null as String?)
        task
    }

    fun newGoogleTask(vararg properties: PropertyValue<in GoogleTask?, *>): GoogleTask {
        return make(instantiator, *properties)
    }
}